xdsl/frontend/code_generation.py

```diff
@@ -71,6 +72,105 @@ def visit_Assign(self, node: ast.Assign) -> None:
         # TODO: Implement assignemnt in the next patch.
         pass
 
+    def visit_BinOp(self, node: ast.BinOp):
+        op_name: str = node.op.__class__.__name__
+
+        # Table with currently unsupported Python AST operators.
+        unsupported_python_AST_operator = {
+            "BitOr", "BitXor", "Div", "FloorDiv", "Mod", "MatMult", "Pow"
+        }
+
+        # Table with mappings of Python AST operator to Python methods.
+        python_AST_operator_to_python_overload = {
+            "Add": "__add__",
+            "BitAnd": "__and__",
+            "LShift": "__lshift__",
+            "Mult": "__mul__",
+            "RShift": "__rshift__",
+            "Sub": "__sub__",
+        }
+        if op_name in unsupported_python_AST_operator:
+            raise CodeGenerationException(
+                node.lineno, node.col_offset,
```

Comment:
Ah, I think I didn't make my point about this clear enough: Rather than failing here already, the code here should happily accept any possible comparator and only fail later in resolve_op_overload. This way different FrontendTypes can support different sets of binary operators.